### PR TITLE
fix: refresh rfox opportunity balance on stake and unstake

### LINKED ISSUE #7417: rFOX on Earn page needs refresh to update balance
### Overview

When staking fox on rFOX page, the balance updates automatically. However, the balance shown on the earn page only updates after a refresh 

### References and additional details

https://jam.dev/c/80be1c51-cf65-47e6-8934-9885d387e3f2

### Acceptance Criteria

Balance on earn page should update automatically

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
The rFOX opportunity balance was not refreshed after doing actions on the rfox page such as stake and unstake 

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #7417

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing
- Go to the earn page
- Go to rfox page and stake an amount
- You should see the opportunity balance updated on the earn page after while

Same with unstake

### Engineering
n/a

### Operations
n/a

## Screenshots (if applicable)
Stake
https://jam.dev/c/21850fd1-41fc-4ec7-b72e-fb839a5247e8
https://jam.dev/c/84612827-e823-416e-adc7-08f678952e8a

### CODE REVIEW

**Review by gomesalexandre — APPROVED** [review]
Tested locally, confirmed this does what it says on the box!

Added one comment re: doing this in a saner home following the DeFi abstraction if you feel like it

https://jam.dev/c/b875983d-6e2a-4a10-990a-efb2137c6ad5

File: src/pages/RFOX/components/Stake/Stake.tsx [hunk 1]
```diff
@@ -87,13 +98,38 @@ export const StakeRoutes: React.FC<StakeRouteProps> = ({ headerComponent }) => {
...
+              assetReference: fromAssetId(stakingAssetId).assetReference,
+            }),
+            accountId: confirmedQuote.stakingAssetAccountId,
+            defiProvider: DefiProvider.rFOX,
+            defiType: DefiType.Staking,
+          },
+        ],
+        { forceRefetch: true },
+      ),
+    )
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/pages/RFOX/components/Stake/Stake.tsx]: Nah, totally makes sense, I changed it in [06fb17b](https://github.com/shapeshift/web/pull/7421/commits/06fb17b7fe9dc6297f087557c54f31e785224ced)

Thanks for the tip!

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 2]
```diff
@@ -53,8 +60,9 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
     ({ chainId, data, transfers, status, txid }: Transaction, accountId: AccountId) => {
       if (status !== TxStatus.Confirmed) return
 
-      const { getOpportunitiesUserData } = opportunitiesApi.endpoints
+      const { getOpportunitiesUserData, getOpportunityUserData } = opportunitiesApi.endpoints
 
+      const shouldRefetchRfoxOpportunity = data?.parser === 'rfox'
```
Comment (gomesalexandre) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:65]: 🐐

File: src/pages/RFOX/components/Stake/Stake.tsx [hunk 3]
```diff
@@ -87,6 +87,7 @@ export const StakeRoutes: React.FC<StakeRouteProps> = ({ headerComponent }) => {
       : undefined,
     stakingAssetId,
   })
+
```
Comment (gomesalexandre) [inline src/pages/RFOX/components/Stake/Stake.tsx:90]: ser

**Review by gomesalexandre — APPROVED** [review]
Retested locally, still does the do! Get in ser.

https://jam.dev/c/1e40c102-7c86-40c0-8585-f4a3adf9c24c

File: src/pages/RFOX/components/Stake/Stake.tsx (hunk 3 above)
Comment (NeOMakinG) [reply src/pages/RFOX/components/Stake/Stake.tsx]: removed the diff in [bac0207](https://github.com/shapeshift/web/pull/7421/commits/bac0207cf9dfa4c819b0bbdb7c8e4fe82bd2c8ec)

**Pull request merged**